sktime/forecasting/compose/_hierarchy_ensemble.py

```diff
@@ -285,7 +285,7 @@ def _get_node_dict(self, z):
                 if counter == 0:
                     nodes = mi
                 else:
-                    nodes.append(mi)
+                    nodes = nodes.append(mi)
```

Comment:
Is this an intended change? Do you want to change `nodes` to `None` here?

```pycon
>>>
>>> a = []
>>> a = a.append(2)
>>> a
>>> a is None
True
>>>
```

Reply:
Thanks for reviewing. Its an intended change as nodes is not of type list rather pd.Index. I too realized after spotting the bug that append function works little different with pd.Index object.
```python
>>>
>>> a = pd.Index([1,2])
>>> b = pd.Index([3])
>>> a = a.append(b)
>>> a
Index([1, 2, 3], dtype='int64')
>>>
```

Reply:
Thanks for clarifying. In that case, can you please add a comment in the code for next time someone comes at this line? This is not intuitive at all (not to me at least), especially after seeing `nodes = []` in L278.